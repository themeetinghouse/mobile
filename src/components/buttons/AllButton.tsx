import React from 'react';
import { Button, Right, Text, Thumbnail } from 'native-base';
import { StyleSheet, ImageSourcePropType } from 'react-native';

import { Theme } from '../../Theme.style';

const style = StyleSheet.create({
  button: {
    backgroundColor: Theme.colors.black,
    borderColor: Theme.colors.gray2,
    borderBottomWidth: 1,
    borderTopWidth: 1,
    borderLeftWidth: 0,
    borderRightWidth: 0,
    height: 63,
  },
  text: {
    fontFamily: Theme.fonts.fontFamilyRegular,
    fontSize: Theme.fonts.medium,
  },
  icon: {
    width: Theme.icons.width,
    height: Theme.icons.height,
    marginRight: 15,
  },
});

type Params = {
  children: string;
  handlePress?(): void;
  icon?: ImageSourcePropType;
};

export default function AllButton({
  children,
  handlePress,
  icon,
}: Params): JSX.Element {
  return (
    <Button full iconRight style={style.button} onPress={handlePress}>
      <Text uppercase={false} style={style.text}>
        {children}
      </Text>
      <Right>
        <Thumbnail
          square
          source={icon ?? Theme.icons.white.arrow}
          style={style.icon}
        />
      </Right>
    </Button>
  );
}
