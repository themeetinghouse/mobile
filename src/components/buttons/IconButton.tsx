import React from 'react';
import { View, Right, Thumbnail, Text } from 'native-base';
import {
  TouchableOpacity,
  ImageSourcePropType,
  StyleSheet,
  ViewStyle,
  TextStyle,
} from 'react-native';
import { Theme, Style } from '../../Theme.style';

const buttonStyle = StyleSheet.create({
  container: {
    paddingTop: 10,
    paddingBottom: 10,
  },
  button: {
    backgroundColor: Theme.colors.transparent,
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
  },
  icon: {
    marginLeft: 0,
  },
  iconRight: {
    width: Theme.icons.width,
    height: Theme.icons.height,
    marginRight: 15,
  },
  label: {
    borderBottomColor: Theme.colors.white,
    borderBottomWidth: 1,
    paddingTop: 3,
    paddingBottom: 3,
    marginLeft: 15,
    marginRight: 15,
    color: Theme.colors.white,
  },
});

interface Props {
  style?: ViewStyle;
  labelStyle?: TextStyle;
  onPress?: () => void;
  icon?: ImageSourcePropType;
  label: string;
  rightArrow?: boolean;
}

export default function IconButton({
  style,
  labelStyle,
  onPress,
  icon,
  label,
  rightArrow,
}: Props): JSX.Element {
  return (
    <View style={[buttonStyle.container, style]}>
      <TouchableOpacity style={buttonStyle.button} onPress={onPress}>
        {icon && (
          <Thumbnail
            square
            source={icon}
            style={[Style.icon, buttonStyle.icon]}
          />
        )}
        <View>
          <Text style={[buttonStyle.label, (style && labelStyle) || {}]}>
            {label}
          </Text>
        </View>
        {rightArrow && (
          <Right>
            <Thumbnail
              source={Theme.icons.white.arrow}
              style={buttonStyle.iconRight}
            />
          </Right>
        )}
      </TouchableOpacity>
    </View>
  );
}
