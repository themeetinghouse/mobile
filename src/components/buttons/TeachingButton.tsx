import React from 'react';
import { View, Button, Text, Thumbnail } from 'native-base';
import { ViewStyle, ImageSourcePropType, StyleSheet } from 'react-native';
import Theme from '../../Theme.style';

const style = StyleSheet.create({
  button: {
    backgroundColor: Theme.colors.white,
    borderRadius: 0,
    height: '100%',
  },
  label: {
    color: Theme.colors.black,
    fontFamily: Theme.fonts.fontFamilyBold,
    fontSize: Theme.fonts.medium,
    paddingRight: 0,
    paddingTop: 4,
  },
  buttonInactive: {
    backgroundColor: Theme.colors.grey2,
    borderRadius: 0,
    height: '100%',
  },
  labelInactive: {
    color: 'white',
    fontFamily: Theme.fonts.fontFamilyBold,
    fontSize: Theme.fonts.medium,
    paddingRight: 0,
    paddingTop: 4,
  },
  icon: {
    width: 24,
    height: 24,
  },
});

interface Params {
  wrapperStyle?: ViewStyle;
  label: string;
  onPress?: () => void;
  active: boolean;
  iconActive: ImageSourcePropType;
  iconInactive: ImageSourcePropType;
}

export default function TeachingButton({
  wrapperStyle,
  label,
  onPress,
  active,
  iconActive,
  iconInactive,
}: Params): JSX.Element {
  return (
    <View style={wrapperStyle}>
      <Button
        style={active ? style.button : style.buttonInactive}
        block
        onPress={onPress}
      >
        <Thumbnail
          square
          source={active ? iconActive : iconInactive}
          style={style.icon}
        />
        <Text
          style={active ? style.label : style.labelInactive}
          uppercase={false}
        >
          {label}
        </Text>
      </Button>
    </View>
  );
}
