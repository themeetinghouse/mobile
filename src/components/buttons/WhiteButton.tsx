import React from 'react';
import { View, Button, Text } from 'native-base';
import Theme from '../../Theme.style';
import { ViewStyle, StyleSheet, ActivityIndicator } from 'react-native';

const styles = StyleSheet.create({
  button: {
    backgroundColor: Theme.colors.white,
    borderRadius: 0,
    height: '100%',
  },
  label: {
    color: Theme.colors.black,
    fontFamily: Theme.fonts.fontFamilyBold,
    fontSize: Theme.fonts.medium,
  },
  buttonBlack: {
    backgroundColor: 'black',
    borderRadius: 0,
    height: '100%',
  },
  labelBlack: {
    color: 'white',
    fontFamily: Theme.fonts.fontFamilyBold,
    fontSize: Theme.fonts.medium,
  },
  buttonOutlined: {
    backgroundColor: 'transparent',
    borderRadius: 0,
    height: '100%',
    borderWidth: 3,
    borderColor: 'white',
  },
  labelOutlined: {
    color: 'white',
    fontFamily: Theme.fonts.fontFamilyBold,
    fontSize: Theme.fonts.medium,
  },
});

interface Params {
  style?: ViewStyle;
  label: string;
  onPress?: () => any;
  outlined?: boolean;
  solidBlack?: boolean;
}

export default function WhiteButton({
  style,
  label,
  onPress,
  outlined,
  solidBlack,
}: Params): JSX.Element {
  return (
    <View style={style}>
      <Button
        style={
          outlined
            ? styles.buttonOutlined
            : solidBlack
            ? styles.buttonBlack
            : styles.button
        }
        block
        onPress={onPress}
      >
        <Text
          style={
            outlined
              ? styles.labelOutlined
              : solidBlack
              ? styles.labelBlack
              : styles.label
          }
          uppercase={false}
        >
          {label}
        </Text>
      </Button>
    </View>
  );
}

interface AsyncParams extends Params {
  isLoading: boolean;
}

// eslint-disable-next-line react/prop-types
export function WhiteButtonAsync({
  style,
  label,
  onPress,
  outlined,
  solidBlack,
  isLoading,
}: AsyncParams): JSX.Element {
  return (
    <View style={style}>
      <Button
        disabled={isLoading}
        style={
          outlined
            ? styles.buttonOutlined
            : solidBlack
            ? styles.buttonBlack
            : styles.button
        }
        block
        onPress={onPress}
      >
        {isLoading ? (
          <ActivityIndicator color="black" />
        ) : (
          <Text
            style={
              outlined
                ? styles.labelOutlined
                : solidBlack
                ? styles.labelBlack
                : styles.label
            }
            uppercase={false}
          >
            {label}
          </Text>
        )}
      </Button>
    </View>
  );
}
