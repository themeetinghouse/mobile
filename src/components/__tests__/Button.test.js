import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import AllButton from '../buttons/AllButton';
import IconButton from '../buttons/IconButton';
import TeachingButton from '../buttons/TeachingButton';
import WhiteButton, { WhiteButtonAsync } from '../buttons/WhiteButton';

const mockPress = jest.fn();

afterEach(() => {
  mockPress.mockClear();
});

describe('Snapshot and functional tests for button components', () => {
  test('All button, default props', () => {
    const { getByText, toJSON } = render(
      <AllButton handlePress={mockPress}>Test</AllButton>
    );

    const button = getByText('Test');
    fireEvent.press(button);
    fireEvent.press(button);

    expect(mockPress).toBeCalledTimes(2);

    expect(toJSON()).toMatchSnapshot();
  });

  test('Icon button, default props', () => {
    const { getByText, toJSON } = render(
      <IconButton onPress={mockPress} label="Testing" />
    );

    const button = getByText('Testing');
    fireEvent.press(button);

    expect(mockPress).toBeCalledTimes(1);

    expect(toJSON()).toMatchSnapshot();
  });

  test('Icon button, test rightArrow prop', () => {
    const { toJSON } = render(<IconButton onPress={mockPress} rightArrow />);

    expect(toJSON()).toMatchSnapshot();
  });

  test('Teaching button, active', () => {
    const { toJSON, getByText } = render(
      <TeachingButton onPress={mockPress} active label="TMH" />
    );

    const button = getByText('TMH');
    fireEvent.press(button);

    expect(mockPress).toBeCalledTimes(1);

    expect(toJSON()).toMatchSnapshot();
  });

  test('Teaching button, inactive', () => {
    const { toJSON, getByText } = render(
      <TeachingButton onPress={mockPress} active={false} label="TMH" />
    );

    const button = getByText('TMH');
    for (let i = 0; i < 50; i++) {
      fireEvent.press(button);
    }

    expect(mockPress).toBeCalledTimes(50);

    expect(toJSON()).toMatchSnapshot();
  });

  test('White button, outlined', () => {
    const { toJSON, getByText } = render(
      <WhiteButton onPress={mockPress} label="TMH" outlined />
    );

    const button = getByText('TMH');
    for (let i = 0; i < 12; i++) {
      fireEvent.press(button);
    }

    expect(mockPress).toBeCalledTimes(12);

    expect(toJSON()).toMatchSnapshot();
  });

  test('White button, solidBlack', () => {
    const { toJSON, getByText } = render(
      <WhiteButton label="TMH" solidBlack />
    );

    // button label, uppercase prop is false
    expect(getByText('TMH').props.uppercase).toBeFalsy();

    expect(toJSON()).toMatchSnapshot();
  });

  test('Async white button, outlined, loading', () => {
    const { toJSON, queryByText } = render(
      <WhiteButtonAsync onPress={mockPress} label="TMH" outlined isLoading />
    );

    expect(queryByText('TMH')).toBeFalsy();
    expect(mockPress).toBeCalledTimes(0);

    expect(toJSON()).toMatchSnapshot();
  });

  test('Async white button, solidBlack, not loading', () => {
    const { toJSON, getByText } = render(
      <WhiteButtonAsync label="TMH" solidBlack isLoading={false} />
    );

    // button label, uppercase prop is false
    expect(getByText('TMH').props.uppercase).toBeFalsy();

    expect(toJSON()).toMatchSnapshot();
  });
});
