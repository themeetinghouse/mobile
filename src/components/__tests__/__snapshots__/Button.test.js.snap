// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot and functional tests for button components All button, default props 1`] = `
<View
  accessible={true}
  focusable={true}
  forwardedRef={[Function]}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "backgroundColor": "#000000",
      "borderBottomWidth": 1,
      "borderColor": "#1a1a1a",
      "borderLeftWidth": 0,
      "borderRadius": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 1,
      "borderWidth": undefined,
      "elevation": 2,
      "flexDirection": "row",
      "height": 63,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 6,
      "paddingTop": 6,
      "shadowColor": undefined,
      "shadowOffset": undefined,
      "shadowOpacity": undefined,
      "shadowRadius": undefined,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "color": "#fff",
          "fontFamily": "System",
          "fontSize": 16.5,
          "marginLeft": 0,
          "marginRight": 0,
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        Object {
          "fontFamily": "Graphik-Regular-App",
          "fontSize": 16,
        },
      ]
    }
    uppercase={false}
  >
    Test
  </Text>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "alignSelf": "center",
        "flex": 1,
      }
    }
  >
    <Image
      source={1}
      square={true}
      style={
        Array [
          Object {
            "borderRadius": 0,
            "height": 56,
            "width": 56,
          },
          Object {
            "height": 24,
            "marginRight": 15,
            "width": 24,
          },
        ]
      }
    />
  </View>
</View>
`;

exports[`Snapshot and functional tests for button components Async white button, outlined, loading 1`] = `
<View
  style={Object {}}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "white",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": 3,
        "elevation": 2,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="black"
      hidesWhenStopped={true}
      size="small"
    />
  </View>
</View>
`;

exports[`Snapshot and functional tests for button components Async white button, solidBlack, not loading 1`] = `
<View
  style={Object {}}
>
  <View
    accessible={true}
    focusable={false}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "black",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "color": "white",
            "fontFamily": "Graphik-Bold-App",
            "fontSize": 16,
          },
        ]
      }
      uppercase={false}
    >
      TMH
    </Text>
  </View>
</View>
`;

exports[`Snapshot and functional tests for button components Icon button, default props 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "paddingBottom": 10,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "opacity": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "borderBottomColor": "#ffffff",
              "borderBottomWidth": 1,
              "color": "#ffffff",
              "marginLeft": 15,
              "marginRight": 15,
              "paddingBottom": 3,
              "paddingTop": 3,
            },
            Object {},
          ]
        }
        uppercase={false}
      >
        Testing
      </Text>
    </View>
  </View>
</View>
`;

exports[`Snapshot and functional tests for button components Icon button, test rightArrow prop 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "paddingBottom": 10,
        "paddingTop": 10,
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "opacity": 1,
      }
    }
  >
    <View
      style={Object {}}
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "borderBottomColor": "#ffffff",
              "borderBottomWidth": 1,
              "color": "#ffffff",
              "marginLeft": 15,
              "marginRight": 15,
              "paddingBottom": 3,
              "paddingTop": 3,
            },
            Object {},
          ]
        }
        uppercase={false}
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "alignSelf": "center",
          "flex": 1,
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "borderRadius": 28,
              "height": 56,
              "width": 56,
            },
            Object {
              "height": 24,
              "marginRight": 15,
              "width": 24,
            },
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Snapshot and functional tests for button components Teaching button, active 1`] = `
<View
  style={Object {}}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#ffffff",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Image
      square={true}
      style={
        Array [
          Object {
            "borderRadius": 0,
            "height": 56,
            "width": 56,
          },
          Object {
            "height": 24,
            "width": 24,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "color": "#000000",
            "fontFamily": "Graphik-Bold-App",
            "fontSize": 16,
            "paddingRight": 0,
            "paddingTop": 4,
          },
        ]
      }
      uppercase={false}
    >
      TMH
    </Text>
  </View>
</View>
`;

exports[`Snapshot and functional tests for button components Teaching button, inactive 1`] = `
<View
  style={Object {}}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "#1a1a1a",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Image
      square={true}
      style={
        Array [
          Object {
            "borderRadius": 0,
            "height": 56,
            "width": 56,
          },
          Object {
            "height": 24,
            "width": 24,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "color": "white",
            "fontFamily": "Graphik-Bold-App",
            "fontSize": 16,
            "paddingRight": 0,
            "paddingTop": 4,
          },
        ]
      }
      uppercase={false}
    >
      TMH
    </Text>
  </View>
</View>
`;

exports[`Snapshot and functional tests for button components White button, outlined 1`] = `
<View
  style={Object {}}
>
  <View
    accessible={true}
    focusable={true}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "transparent",
        "borderBottomWidth": null,
        "borderColor": "white",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": 3,
        "elevation": 2,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "color": "white",
            "fontFamily": "Graphik-Bold-App",
            "fontSize": 16,
          },
        ]
      }
      uppercase={false}
    >
      TMH
    </Text>
  </View>
</View>
`;

exports[`Snapshot and functional tests for button components White button, solidBlack 1`] = `
<View
  style={Object {}}
>
  <View
    accessible={true}
    focusable={false}
    forwardedRef={[Function]}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "backgroundColor": "black",
        "borderBottomWidth": null,
        "borderColor": "#007aff",
        "borderLeftWidth": null,
        "borderRadius": 0,
        "borderRightWidth": null,
        "borderTopWidth": null,
        "borderWidth": undefined,
        "elevation": 2,
        "flexDirection": "row",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 6,
        "paddingTop": 6,
        "shadowColor": undefined,
        "shadowOffset": undefined,
        "shadowOpacity": undefined,
        "shadowRadius": undefined,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 16.5,
            "marginLeft": 0,
            "marginRight": 0,
            "paddingLeft": 16,
            "paddingRight": 16,
          },
          Object {
            "color": "white",
            "fontFamily": "Graphik-Bold-App",
            "fontSize": 16,
          },
        ]
      }
      uppercase={false}
    >
      TMH
    </Text>
  </View>
</View>
`;
